jwst/extract_2d/grisms.py

```diff
@@ -206,16 +223,21 @@ def extract_tso_object(input_model,
                         input_model.meta.wcsinfo.dispersion_direction
             if compute_wavelength:
                 output_model.wavelength = compute_wavelength_array(output_model)
-            output_model.name = 'TSO object'
+            output_model.name = '1'
+            output_model.source_type = input_model.meta.target.source_type
+            output_model.source_name = input_model.meta.target.catalog_name
+            output_model.source_alias = input_model.meta.target.proposer_name
             output_model.xstart = 1  # fits pixels
```

Comment:
Is xstart always 1 for extracted TSO grism data?

Reply:
The code computes `xmin`, but the entire row is extracted (for NIRCam TSO).  Here's an example:

    ext_data = input_model.data[..., ymin: ymax + 1, :].copy()

So that's why I just specified the value as `1` rather than `xmin + 1`.